packages/console/src/pages/Roles/components/CreateRoleForm/Footer.tsx

```diff
@@ -0,0 +1,121 @@
+import QuotaGuardFooter from '@/components/QuotaGuardFooter';
+import { isCloud, isDevFeaturesEnabled } from '@/consts/env';
+import { TenantsContext } from '@/contexts/TenantsProvider';
+import Button from '@/ds-components/Button';
+import useSubscriptionPlan from '@/hooks/use-subscription-plan';
+import { hasReachedQuotaLimit } from '@/utils/quota';
+import { buildUrl } from '@/utils/url';
+
+type Props = {
+  roleType: RoleType;
+  selectedScopesCount: number;
+  isCreating: boolean;
+  onClickCreate: () => void;
+};
+
+function Footer({ roleType, selectedScopesCount, isCreating, onClickCreate }: Props) {
+  const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
+  const { currentTenantId } = useContext(TenantsContext);
+  const { data: currentPlan } = useSubscriptionPlan(currentTenantId);
+  const { data: [, roleCount] = [] } = useSWR<[RoleResponse[], number]>(
+    isCloud &&
+      buildUrl('api/roles', {
+        page: String(1),
+        page_size: String(1),
+        type: roleType,
+      })
+  );
+
+  const isRoleReachLimit = hasReachedQuotaLimit({
+    quotaKey: roleType === RoleType.User ? 'rolesLimit' : 'machineToMachineRolesLimit',
+    plan: currentPlan,
+    usage: roleCount ?? 0,
+  });
+
+  const isScopesPerRoleReachLimit = hasReachedQuotaLimit({
+    quotaKey: 'scopesPerRoleLimit',
+    plan: currentPlan,
+    /**
+     * If usage is equal to the limit, it means the current role has reached the maximum allowed scope.
+     * Therefore, we should not assign any more scopes at this point.
+     * However, the currently selected scopes haven't been assigned yet, so we subtract 1
+     * to allow the assignment when the scope count is equal to the limit.
+     */
+    usage: selectedScopesCount - 1,
+  });
+
+  if (currentPlan) {
+    if (isRoleReachLimit) {
+      if (roleType === RoleType.User) {
+        return (
+          <QuotaGuardFooter>
+            <Trans
+              components={{
+                a: <ContactUsPhraseLink />,
+                planName: <PlanName name={currentPlan.name} />,
+              }}
+            >
+              {t('upsell.paywall.roles', { count: currentPlan.quota.rolesLimit ?? 0 })}
+            </Trans>
+          </QuotaGuardFooter>
+        );
+      }
+
+      // RoleType === RoleType.MachineToMachine
+      return (
+        <QuotaGuardFooter>
+          <Trans
+            components={{
+              a: <ContactUsPhraseLink />,
+              planName: <PlanName name={currentPlan.name} />,
+            }}
+          >
+            {/* Todo @xiaoyijun [Pricing] Remove feature flag */}
+            {!isDevFeaturesEnabled && currentPlan.id === ReservedPlanId.Free
+              ? t('upsell.paywall.deprecated_machine_to_machine_feature')
+              : t('upsell.paywall.machine_to_machine_roles', {
+                  count: currentPlan.quota.machineToMachineRolesLimit ?? 0,
+                })}
+          </Trans>
+        </QuotaGuardFooter>
```

Comment:
Write the condition in the children of the `<Trans>` component maybe, to avoid having redundant code

Reply:
Good call, updated.